glean_parser/schemas/metrics.2-0-0.schema.yaml

```diff
@@ -625,3 +628,36 @@ additionalProperties:
               description: |
                 `denominator_metric` is only allowed for `rate`.
               maxLength: 0
+      -
+        if:
+          properties:
+            type:
+              const: text
+        then:
+          properties:
+            lifetime:
+              description: |
+                Text metrics must have ping or application lifetime.
+              enum:
+                - ping
+                - application
+
+            data_sensitivity:
+              description: |
+                Text metrics require Category 3 (`web_activity`) or Category 4 (`highly_sensitive`).
+              type: array
+              items:
+                enum:
+                  - web_activity
+                  - highly_sensitive
+
+            send_in_pings:
+              description: |
+                Text metrics can only be sent in custom pings. Built-in pings are not allowed.
+              type: array
+              items:
+                allOf:
+                  - $ref: "#/definitions/kebab_case"
+                  - not:
+                      description: "Text metrics can only be sent in custom pings. Built-in pings are not allowed."
+                      pattern: "^(metrics|baseline|events|deletion-request|default|glean_.*)$"
```

Comment:
Hum, TIL about `default`. Is that a keyword someone can add in order to get the default ping for the metric type? If so it should be documented [here](https://mozilla.github.io/glean/book/reference/yaml/metrics.html#send_in_pings) and should also be disallowed as a ping name (I just checked that it is not).

These are all follow up stuff though.

Reply:
hm, I think so? https://github.com/mozilla/glean_parser/blob/text-metric/glean_parser/metrics.py#L80-L81

Reply:
Yeah it is also mentioned [here](https://mozilla.github.io/glean/book/user/pings/custom.html#sending-metrics-in-a-custom-ping) (found this after sending the review). Either way, good to document it in the metrics parameters as well.

Reply:
https://github.com/mozilla/glean_parser/blob/text-metric/glean_parser/parser.py#L337-L340

Reply:
> https://github.com/mozilla/glean_parser/blob/text-metric/glean_parser/parser.py#L337-L340

Yeah, this checks the `send_in_pings` parameter, but the ping name is not checked. Currently `default` is an accepted ping name. Of course this is a minor bug, but still.

Reply:
Right, I'll fix that separately.